Modules/Microsoft365DSC/Modules/M365DSCUtil.psm1

```diff
@@ -3361,38 +3361,50 @@ function Get-M365DSCExportContentForResource
     $Results = Format-M365DSCString -Properties $Results `
         -ResourceName $ResourceName
 
+    if ($Script:AllM365DscResources.Count -eq 0)
+    {
+        $Script:AllM365DscResources = Get-DscResource -Module 'Microsoft365Dsc'
+    }
+
     $primaryKey = ''
-    if ($Results.ContainsKey('IsSingleInstance'))
+    $Resource = $Script:AllM365DscResources.Where({ $_.Name -eq  $ResourceName })
+    $Keys = $Resource.Properties.Where({ $_.IsMandatory }) | `
```

Comment:
As an alternative approach we could remove IsSingleInstance here and iterate over all keys and grab the values of the $Result list this:

PseudoCode:

$keys.Remove("IsSingleInstance")
$primaryKeys = @()
foreach($key in $keys){$primaryKeys += $Results.$Key}

Reply:
For resources that are IsSingleInstance the $primaryKey is IsSingleInstance itself but we want to end up with ResourceInstanceName equal to the name of the Resource itself, so this wouldn't work. what I additionally do is for resources that have multiple mandatory keys but are not the usual ones I join them into a single string so you'd get something like "ResourceName-ValueOfKey1ValueOfKey2ValueOfKey3", and for all others just "ResourceName-ValueOfKey1".

Reply:
Thanks for the clarification.  Just read lines 3395 to 3407. This makes sense. Please ignore my comments then. :)

Reply:
One question left, will we have unique Names for all Resources with this approach?

Reply:
That's a side effect of what I wanted to achieve initially but yes that's the intention, with the code that it's there right now there's no 100% guarantee that happens.

I don't think the current unit tests are useful to test this change but I exported the whole Intune workload on 3 separate tenants, using this code, without issues and all resources got unique ResourceInstanceNames.

Reply:
... This would only apply in a scenario of a well setup tenant with unique names. All resources, that require multiple keys are only unique, if we use all of the keys. Or will this over-complicate things?

Reply:
Ideally we will always want each resource to be unique or we'll have problems, so except for the IsSingleInstance case then yes in order to be totally unique would be to join all keys. Nevertheless for now I think this would be already better compared to what's there now, so if it's approved let it settle for a few weeks and if no one complains then we can change to join all mandatory keys except for resources that are IsSingleInstance since as previously mentioned those ones should end up with ResourceInstanceName = ResourceName.

Reply:
Just did another review of the code and this change will be the best option and does not require any further changes, as there is  a further check for uniqueness in L3421+ which will mitigate possible duplicates and we would still end up with the best name and a numeric value in the end.